### PR TITLE
Issue #375 - Add jar classifiers to the dependency cache.

### PR DESCRIPTION
**Pull request by cwoodwar6**

### CODE REVIEW

**Comment by cwoodwar6** [comment]
Blocked on https://github.com/cla-assistant/cla-assistant/issues/198

File: buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/ExternalDependency.groovy [hunk 1]
```diff
@@ -23,6 +23,10 @@ class ExternalDependency extends VersionlessDependency {
         }
 
         this.depFile = depFile
+
+        if (artifactType == "jar") {
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/ExternalDependency.groovy]: why do we need to filter here?

Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/ExternalDependency.groovy]: I think the check should be if `artifactClassifier` exists, assign it

File: buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/VersionlessDependency.groovy [hunk 2]
```diff
@@ -10,13 +10,18 @@ class VersionlessDependency {
     final String group
     final String name
 
+    String classifier
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/VersionlessDependency.groovy]: Make this final and pass in default null

File: buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/ExternalDependency.groovy (hunk 1 above)
Comment (cwoodwar6) [inline buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/ExternalDependency.groovy]: When I was testing this, it included AARs. Those AARs had classifiers such as "debug" which seemed to break dependency resolution. This was just to keep it simple and not have to make any additional changes.

File: buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/VersionlessDependency.groovy (hunk 2 above)
Comment (cwoodwar6) [inline buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/VersionlessDependency.groovy]: Done. This change adds the additional arguments to the constructor and pushes off the type logic here as well.

File: buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/VersionlessDependency.groovy [hunk 3]
```diff
@@ -9,14 +9,21 @@ class VersionlessDependency {
 
     final String group
     final String name
+    final String classifier
 
-    VersionlessDependency(ModuleVersionIdentifier identifier) {
+    VersionlessDependency(ModuleVersionIdentifier identifier, String artifactType, String artifactClassifier) {
         group = identifier.group
         name = identifier.name
+        if (artifactType == "jar") {
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/VersionlessDependency.groovy]: This is q very custom hack that only works for jar artifacts. I do not like how this is currently setup. aars can have classifiers as well when they are external dependencies.

I think the proper fix is to ignore the artifactType and classifier if it is not an external dependency.

**Comment by kageiit** [comment]
Superceded by @388

**Pull request closed without merging**